### PR TITLE
Add filename to flow descriptions

### PR DESCRIPTION
**Pull request by Andre0991**
- Add filename to flow descriptions. Use the following format: `<description> (<file>:<line>)`
- Put a map with flow metadata (line, file, etc) in state instead of directly putting the formatted string.

### CODE REVIEW

File: src/state_flow/core.clj [hunk 1]
```diff
@@ -31,13 +31,16 @@
...
-  [description {:keys [line]}]
+  [description {:keys [line ns file]}]
   (modify-meta
    (fn [m] (-> m
                (update :top-level-description #(or % description))
-               (update :description-stack (fnil conj []) (str description
-                                                              (when line
-                                                                (format " (line %s)" line))))))))
+               (update :description-stack (fnil conj [])
+                       ;; TODO: Add filename as well
```
Comment (sovelten) [inline src/state_flow/core.clj:39]: remove todo

File: src/state_flow/assertions/matcher_combinators.clj [hunk 2]
```diff
@@ -52,12 +52,14 @@
...
    ;; caller-meta is definitely not part of the API.
-  (let [caller-meta      (meta &form)
-        params*          (merge {:description  "match?"
-                                 :caller-meta  caller-meta
-                                 :times-to-try 1
-                                 :sleep-time   probe/default-sleep-time}
-                                params)]
+  (let [f           *file*
+        a-ns        (str *ns*)
+        caller-meta (assoc (meta &form) :file f :ns a-ns)
```
Comment (dchelimsky) [inline src/state_flow/assertions/matcher_combinators.clj:57]: 💅 I'd prefer inlining e.g. `(assoc (meta &form) :file *file* :ns *ns*)`, and maybe even inlining the caller-meta in params* below.

File: src/state_flow/core.clj [hunk 3]
```diff
@@ -31,13 +31,15 @@
...
-  [description {:keys [line]}]
+  [description {:keys [line ns file]}]
   (modify-meta
    (fn [m] (-> m
                (update :top-level-description #(or % description))
-               (update :description-stack (fnil conj []) (str description
-                                                              (when line
-                                                                (format " (line %s)" line))))))))
+               (update :description-stack (fnil conj [])
+                       (merge {:description description
```
Comment (dchelimsky) [inline src/state_flow/core.clj:39]: I know how big a fan of `cond->` you are, so I'll just throw this out as a suggestion with a weak preference:

```clojure
(cond-> {:description description :ns ns)
  line (assoc :line line)
  file (assoc :file file))
```

File: src/state_flow/core.clj [hunk 4]
```diff
@@ -118,9 +129,13 @@
...
   {:style/indent :defn}
   [description & flows]
-  (apply flow* {:description description
-                :caller-meta (meta &form)}
-         flows))
+  (let [f    *file*
+        a-ns (str *ns*)]
+    (apply flow* {:description description
+                  :caller-meta (assoc (meta &form)
+                                      :file f
```
Comment (dchelimsky) [inline src/state_flow/core.clj:136]: 💅 inline `f` and `a-ns`

**Review by dchelimsky — APPROVED** [review]

File: src/state_flow/core.clj (hunk 3 above)
Comment (sovelten) [reply src/state_flow/core.clj]:
```clojure
(assoc-some {:description description :ns ns}
  :line line
  :file file)
```
would be the ideal solution in my opinion, but not part of the core. I have my reasons for avoiding arrow macro shenanigans, but will let @Andre0991 decide :P

**Review by sovelten — APPROVED** [review]

**Pull request merged**